### PR TITLE
WIP: 5345: Nonpersistent EJB timers

### PR DESCRIPTION
**Pull request by poikilotherm**
WORK IN PROGRESS - DO NOT MERGE

## Related Issues

- connects to #5345: Proposal: make persistent EJB Timers non-persistent

## Pull Request Checklist

- [ ] Unit [tests][] completed
- [ ] Integration [tests][]: None
- [ ] Deployment requirements, [SQL updates][], [Solr updates][], etc.: None
- [x] [Documentation][docs] completed
- [ ] Merged latest from "develop" [branch][] and resolved conflicts

[tests]: http://guides.dataverse.org/en/latest/developers/testing.html
[SQL updates]: http://guides.dataverse.org/en/latest/developers/sql-upgrade-scripts.html
[Solr updates]: https://github.com/IQSS/dataverse/blob/develop/conf/solr/7.3.0/schema.xml
[docs]: http://guides.dataverse.org/en/latest/developers/documentation.html
[branch]: http://guides.dataverse.org/en/latest/developers/branching-strategy.html

### CODE REVIEW

File: doc/sphinx-guides/source/developers/timers.rst [hunk 1]
```diff
@@ -0,0 +1,19 @@
+==========
+EJB Timers
+==========
+
+As described in :doc:`../admin/timers`, Dataverse uses EJB timers for scheduled jobs. This section is about the
+techniques used for scheduling.
+
+* :doc:`../admin/metadataexport` is done via ``@Schedule`` annotation on ``OAISetServiceBean.exportAllSets()`` and
```
Comment (landreev) [inline doc/sphinx-guides/source/developers/timers.rst]: My main question/source of confusion about this PR so far: I'm not seeing the @Schedule annotation mentioned here in either OAISetServiceBean or DatasetServiceBean in your branch. Am I missing something?

File: doc/sphinx-guides/source/admin/timers.rst [hunk 2]
```diff
@@ -3,50 +3,61 @@
...
+  ``<ejb-timer-service timer-datasource="jdbc/VDCNetDS">`` is changed back to ``<ejb-timer-service>``
+  on all the non-timer server nodes. Similarly, this option is automatically set by the installer script.
+  Changing it back to the default setting on a server that doesn't need to run the timer will prevent a potential
+  race condition, where multiple servers try to get a lock on the timer database.
 
 Known Issues
 ------------
  
-We've received several reports of an intermittent issue where the application fails to deploy with the error message "EJB Timer Service is not available." Please see the :doc:`/admin/troubleshooting` section of this guide for a workaround. 
+Former to Dataverse 4.10, we've received several reports of an intermittent issue where the application fails to deploy
```
Comment (landreev) [inline doc/sphinx-guides/source/admin/timers.rst]: "Prior to Dataverse 4.10" probably?

File: doc/sphinx-guides/source/developers/timers.rst (hunk 1 above)
Comment (poikilotherm) [reply doc/sphinx-guides/source/developers/timers.rst]: I just pushed actual codework on this. Until now, this was only documentation work, as @pdurbin suggested documentation-driven-development... :wink:

File: doc/sphinx-guides/source/developers/timers.rst [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  ``DatasetServiceBean.exportAll()``. Fixed to 2AM local time every day, non persistent.
+* Harvesting is a bit more complicated. The timer is attached to ``HarvesterServiceBean.harvestEnabled()`` via
+  ``@Schedule`` annotation every hour at minute 50, non-persistent.
```
[the remark this answers is not in the document]
Comment (poikilotherm) [reply doc/sphinx-guides/source/developers/timers.rst]: Yeah, see my comment above: this needs to be done now.

File: doc/sphinx-guides/source/admin/timers.rst [hunk 4]
```diff
@@ -3,50 +3,61 @@
...
+``asadmin delete-system-property "dataverse.timerServer"``.
+(This option will not be set to ``true`` in future Docker images of Dataverse, it needs to be configured.)
 
-This daily job will also update all the harvestable OAI sets configured on your server, adding new and/or newly published datasets or marking deaccessioned datasets as "deleted" in the corresponding sets as needed. 
+As we don't use persistent timers from Dataverse 4.10 onward, when upgrading, it is up to you to follow the former
+recommendation or not. In new installations, this will not be necessary.
 
-This job is automatically scheduled to run at 2AM local time every night. If really necessary, it is possible (for an advanced user) to change that time by directly editing the EJB timer application table in the database.  
+  We also recommend that the following entry in the :fixedwidthplain:`domain.xml`:
+  ``<ejb-timer-service timer-datasource="jdbc/VDCNetDS">`` is changed back to ``<ejb-timer-service>``
```
[the remark this answers is not in the document]
Comment (poikilotherm) [reply doc/sphinx-guides/source/admin/timers.rst]: Just to be sure we are talking about the same: this instruction has been included in the docs already. Once this PR is merged, IMHO the described steps are not necessary anymore.

A blank option does not mean "non-persistent" by default, a blank option means any new *persistent* timers introduced later on will be stored in the default datasource (local H2 or Hazelcast cache in Payara). (You need to explicitly declare timers as "non-persistent".)

Yes, the installer (and other places) should be changed, too. The `asadmin` commands from `setup-glassfish.sh` regarding JDBC timers can be pruned. Will put that in a separate commit to have more logical chunks.

**Comment by pdurbin** [comment]
@poikilotherm heads up that this pull request has merge conflicts.

**Comment by poikilotherm** [comment]
Closing in favor of #7416

**Pull request closed without merging**